src/enum.cr

```diff
@@ -275,8 +275,16 @@ struct Enum
   end
 
   # Returns a hash value. This is the hash of the underlying value.
+  # Enum hash is exception from random seeding, cause several Hashes are
+  # initialized on very early stages of bootstrap, and StdHasher seed is not
+  # filled yet.
```

Comment:
You might be able to work around this by adjusting the require order in the prelude.